src/control/cmd/dmg/pool.go

```diff
@@ -193,8 +195,42 @@ func (d *PoolGetACLCmd) Execute(args []string) error {
 		return err
 	}
 
-	d.log.Infof("Pool-get-ACL command succeeded, UUID: %s\n", d.UUID)
-	d.log.Info(formatACL(resp.ACL))
+	d.log.Debugf("Pool-get-ACL command succeeded, UUID: %s\n", d.UUID)
+
+	acl := formatACL(resp.ACL)
+
+	if d.File != "" {
+		err = d.writeACLToFile(acl)
+		if err != nil {
+			return err
+		}
+		d.log.Infof("Wrote ACL to output file: %s", d.File)
+	} else {
+		d.log.Info(acl)
+	}
+
+	return nil
+}
+
+func (d *PoolGetACLCmd) writeACLToFile(acl string) error {
+	// Keep the user from clobbering existing files
+	_, err := os.Stat(d.File)
+	if err == nil {
+		return errors.New(fmt.Sprintf("file already exists: %s", d.File))
+	}
+
+	f, err := os.Create(d.File)
+	if err != nil {
+		d.log.Errorf("Unable to create file: %s", d.File)
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(acl)
```

Comment:
Is it possible that this will ever do a partial write or is it guaranteed to be all or nothing?

Reply:
all or nothing . "Write returns a non-nil error when n != len(b)." behaviour inherited from Write() https://golang.org/pkg/os/#File.WriteString